test/jdk/javax/swing/JFileChooser/FileChooserDisableTest.java

```diff
@@ -0,0 +1,161 @@
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.BorderLayout;
+import java.awt.Component;
+import java.awt.Container;
+import java.awt.Dimension;
+import java.awt.Point;
+import java.awt.Robot;
+import java.awt.event.InputEvent;
+
+import javax.swing.AbstractButton;
+import javax.swing.JButton;
+import javax.swing.JFileChooser;
+import javax.swing.JFrame;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+import javax.swing.WindowConstants;
+import javax.swing.UnsupportedLookAndFeelException;
+
+import java.util.function.Predicate;
+
+/*
+ * @test
+ * @bug 4365952
+ * @key headful
+ * @summary Test to check is File chooser can be Disabled
+ * @run main FileChooserDisableTest
+ */
+public class FileChooserDisableTest {
+    static JFrame frame;
+    static JFileChooser jfc;
+    static volatile Point movePoint;
+    static String buttonToolTip;
+    static volatile AbstractButton openBtn;
+
+    public static void main(String[] args) throws Exception {
+        Robot robot = new Robot();
+
+        for (UIManager.LookAndFeelInfo laf : UIManager.getInstalledLookAndFeels()) {
+            if (laf.getClassName().toLowerCase().contains("motif")) {
+                continue;
+            }
+
+            if (laf.getClassName().toLowerCase().contains("gtk")) {
+                buttonToolTip = "Open selected file.";
+            } else {
+                buttonToolTip = "Open selected file";
+            }
+
+            System.out.println("Testing LAF: " + laf.getClassName());
+            SwingUtilities.invokeAndWait(() -> setLookAndFeel(laf));
+            try {
+                SwingUtilities.invokeAndWait(() -> {
+                    initialize();
+                });
+                String defaultDirectory = jfc.getCurrentDirectory().toString();
+                robot.delay(1000);
+                robot.waitForIdle();
+                SwingUtilities.invokeAndWait(() -> {
+                    openBtn = clickOpenButton();
+                    movePoint = openBtn.getLocationOnScreen();
+                });
+                Dimension btnSize = openBtn.getSize();
+                robot.mouseMove(movePoint.x + btnSize.width / 2, movePoint.y + btnSize.height / 2);
+                robot.delay(100);
+                robot.waitForIdle();
+                robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+                robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+                robot.delay(100);
+                robot.waitForIdle();
+                String currentDirectory = jfc.getCurrentDirectory().toString();
+                if (!currentDirectory.equals(defaultDirectory)) {
+                    throw new RuntimeException("File chooser disable failed");
```

Comment:
Better to add the failed LAF name also while throwing exception.

Reply:
I've added print statement mentioning LAF name at line 72. I hope that would be fine?

Reply:
I missed that statement.. looks ok.